### DISCUSSION TITLE
Cordova plugins with <framework custom="true"> and Capacitor

### DISCUSSION THREAD

**Discussion by please-fill-out-this-field** [ionic-team/capacitor#6248] [Q&A]
Hello!

I'm trying to add a Cordova plugin (namely [cordova-social-vk](https://npmjs.com/package/cordova-social-vk)) to my capacitor 3.x.x app.

The plugin contains the following "framework" tag (under android section) in its `plugin.xml`:

```xml
<framework src="src/android/vksdk_library" custom="true" />
```

I checked plugin sources - the directory is there indeed.

But android build always fails - here's the gradle error I'm getting:

```
/path/to/src-capacitor/android/capacitor-cordova-android-plugins/src/main/java/ru/trilan/socialvk/SocialVk.java:29: error: package com.vk.sdk does not exist
import com.vk.sdk.VKAccessToken;
```

I googled and found more people reporting similar issues:

https://forum.ionicframework.com/t/framework-dependency-in-plugin-migration-cordova-capacitor/181146

https://github.com/ionic-team/capacitor/issues/215

But no solutions to try - either they never managed to solve the problem, or decided not to share solutions for some reason.

Anything I can do with the plugin to make it work with Capacitor?

Thanks!

**Comment by jcesarmobile** [comment]
Looks like a bug in Capacitor, but does that plugin work in Cordova apps? 
The plugin has not been updated for almost 6 years, so it's very likely that it doesn't work in Cordova apps neither.

**Reply by please-fill-out-this-field** [reply]
> so it's very likely that it doesn't work in Cordova apps neither.

Indeed.

However, I found a "fixed" version ([here](https://github.com/iRapoo/cordova-vk-auth)) that I confirm is compatible with modern-day cordova, but it just gives me the exact same error.

**Comment by please-fill-out-this-field** [comment]
OP here. Still haven't found a solution.

Any ideas?
